cli/cmd/update.go

```diff
@@ -134,9 +134,13 @@ func updateStackMeta(ctx context.Context, stackName string, happyClient *HappyCl
 		"happy/meta/configsecret": happyClient.HappyConfig.GetSecretId(),
 	})
 	if sliceDefaultTag != "" {
-		happyClient.Tag = sliceDefaultTag
+		happyClient.ArtifactBuilder.WithTags([]string{sliceDefaultTag})
 	}
-	err := stackMeta.Update(ctx, happyClient.Tag, happyClient.StackTags, "", happyClient.StackService)
+	// for updating and creating, there should only be one tag (either provided or generated)
+	if len(happyClient.ArtifactBuilder.GetTags()) != 1 {
+		return nil, errors.New("there should only be one tag when updating or creating a stack")
```

Comment:
If we have a constraint like this, why do we allow a list of tags?

Reply:
ArtifactBuilder is also used in the happy push command and that command allows you to specify multiple tags to push. So we need a way to specify multiple tags. In the update/create case though, you can only specify one tag, so that is why I check it here.